xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -1578,6 +1578,47 @@ void CActiveAE::ChangeResamplers()
   }
 }
 
+static CAEChannelInfo GetInternalChannelLayout(AEStdChLayout stdLayout)
```

Comment:
This code was exactly how it was before the breakage (I've just copied the GetInternalChannelLayout function from ADSP).
I've added a commit to convert directly but the code gets larger (and possibly more likely to be incorrect).